content/tutorial/tutorial.md

```diff
@@ -796,15 +803,15 @@ history = [
 ]
 ```
 
-Now we need to decide which component should own the `history` state.
+अब हमें यह तय करने की आवश्यकता है कि किस कौम्पोनॅन्ट को `history` state का मालिक होना चाहिए।
 
-### Lifting State Up, Again {#lifting-state-up-again}
+### लिफ्टिंग state अप, फिर से {#lifting-state-up-again}
 
-We'll want the top-level Game component to display a list of past moves. It will need access to the `history` to do that, so we will place the `history` state in the top-level Game component.
+हम पिछले चालों की सूची प्रदर्शित करने के लिए शीर्ष-स्तरीय Game कौम्पोनॅन्ट चाहते हैं। यह करने के लिए `history` तक पहुंचने की आवश्यकता होगी, इसलिए हम शीर्ष स्तर के Game कौम्पोनॅन्ट में `history` state को जगह देंगे।
 
-Placing the `history` state into the Game component lets us remove the `squares` state from its child Board component. Just like we ["lifted state up"](#lifting-state-up) from the Square component into the Board component, we are now lifting it up from the Board into the top-level Game component. This gives the Game component full control over the Board's data, and lets it instruct the Board to render previous turns from the `history`.
+Game के कौम्पोनॅन्ट में `history` state को रखने से हम अपने चाइल्ड के Board कौम्पोनॅन्ट से `squares` state को हटा सकते हैं। जैसे हम Board कौम्पोनॅन्ट में ["लिफ़्टेड state अप"](#lifting-state-up) को Square कौम्पोनॅन्ट से हटाते हैं, वैसे ही अब हम इसे Board से टॉप-लेवल Game कौम्पोनॅन्ट में उठा रहे हैं। यह गेम Game कौम्पोनॅन्ट को Board के डेटा पर पूर्ण नियंत्रण प्रदान करता है, और इससे Board को `history` से पिछले घुमावों को प्रस्तुत करने का निर्देश मिलता है।
```

Comment:
जैसे हम Board कौम्पोनॅन्ट में ["लिफ़्टेड state अप"](#lifting-state-up) को Square कौम्पोनॅन्ट se hataya tha

Reply:
Game कौम्पोनॅन्ट में उठा रहे हैं।  -> is okay, but could we have something better, if possible?

Reply:
और इससे Board को `history` से पिछले घुमावों को प्रस्तुत करने का निर्देश मिलता है। -> revisit?

Reply:
> Game कौम्पोनॅन्ट में उठा रहे हैं।

Game कौम्पोनॅन्ट mein la rahe hain?

Reply:
```
और इससे Board को `history` की पिछली चालों को प्रस्तुत करने का निर्देश मिलता है।
```
???